sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -396,6 +396,8 @@ case class Or(left: Expression, right: Expression) extends BinaryOperator with P
 
   override def sqlOperator: String = "OR"
 
+  override def nullable: Boolean = left.nullable && right.nullable
```

Comment:
Based on the semantics of ANSI SQL, the current solution is right. 

```SQL
NULL OR True => True
NULL OR False => NULL
NULL OR NULL => NULL
```

If you also agree on it, could you close it? Thanks!